solutions/usaco-570.mdx

```diff
@@ -2,72 +2,360 @@
+		return;
+	}
+
+	// Set room to visited
+	visited[x][y] = true;
+
+	// Iterate through neighbors and floodfill from them
+	for (int i = 0; i < 4; i++) {
+		floodfill(x + dirX[i], y + dirY[i]);
+	}
+
+	// Turn on all lights from current room
+	for (int i = 0; i < switches[x][y].size(); i++) {
+		int roomX = switches[x][y][i].first;
+		int roomY = switches[x][y][i].second;
 
-	cin >> n >> m;
+		// If room hasn't been lit yet, add it to our count of rooms that have 
+		// been lit
+		if (!illuminated[roomX][roomY]) {
+			litRooms++;
+		}
 
-	light[0][0] = true;
+		// Set room led to by switch to lit
+		illuminated[roomX][roomY] = true;
 
+		// Floodfill from new room that is lit
+		floodfill(roomX, roomY);
+	}
+}
+
+int main() {
+	setIO("lightson");
+
+	int m;
+	cin >> N >> m;
+	
+	// Take in input and add switches to corresponding room
 	for (int i = 0; i < m; i++) {
-		int x1, y1, x2, y2;
-		cin >> x1 >> y1 >> x2 >> y2;
-		x1--; y1--; x2--; y2--;
-		change[y1][x1].push_back({y2, x2});
+		int x, y, a, b;
+		cin >> x >> y >> a >> b;
+		switches[x][y].push_back({a, b});
 	}
 
-	ff(0, 0);
+	// Set the top left room to lit (since it's lit initially)
+	illuminated[1][1] = true;
 
-	cout << sol << '\n';
+	// Start floodfill from top-left room
+	floodfill(1,1);
+
+	cout << litRooms << endl;
 }
 ```
 
 </CPPSection>
+
+<PySection>
+```py
+import sys
+sys.setrecursionlimit(100000)  # Raise recursion limit as the default will error
+
+filein = open("lightson.in",'r')
+N, m = map(int, filein.readline().split())
+
+lit_rooms = 1
+visited = [[False for i in range(N + 1)] for j in range(N + 1)]
+illuminated = [[False for i in range(N + 1)] for j in range(N + 1)]
+switches = [[[] for i in range(N + 1)] for j in range(N + 1)]
+
+# Read in light switches input
+for i in range(m):
+	x, y, a, b = map(int, filein.readline().split())
+	switches[x][y].append((a,b))
+
+# Checks if a room is connected to the main component
+def check_connected(x,y):
+	dir_x = [-1, 0, 1, 0]
+	dir_y= [0, -1, 0, 1]
+
+	# Iterate through neighbors
+	for i in range(4):
+		new_x = x + dir_x[i]
+		new_y = y + dir_y[i]
+
+		# Ignore neighbor if out of bounds
+		if new_x < 1 or new_y < 1 or new_x > N or new_y > N:
+			continue
+		
+		# If a neighbor is visited, the room is connected to the
+		# main component, return true
+		if visited[new_x][new_y]:
+			return True
+	
+	# If no neighbors have been visited, return false
+	return False
+
+# Floodfill method with source room (x,y)
+def floodfill(x,y):
+	global lit_rooms
+
+	# Ignore room if its out of bounds, already visited, or not lit
+	if x < 1 or y < 1 or x > N or y > N or visited[x][y] or not illuminated[x][y]:
+		return
+
+	# Ignore this room if it isn't connected to main component 
+	# (i.e. it's inaccessible to Bessie)
+	# Don't return on the starting coordinate (1,1)
+	if (not check_connected(x,y)) and (not(x == 1 and y == 1)):
+		return
+
+	# Set room to visited
+	visited[x][y] = True
+
+	dir_x = [-1,0,1,0]
+	dir_y = [0,-1,0,1]
```

Comment:
spaces for commas and stuff
there's some other places where this happens as well